src/scripts/enhanced_lifestyle_analyses/enhanced_lifestyle_analyses.py

```diff
@@ -1,10 +1,834 @@
+            'li_mar_stat': ['Never Married', 'Married', 'Divorced or Widowed'],
+            'li_ed_lev': ['No education', 'Primary edu', 'secondary education']
+        }
+
+        # create a dictionary to defining individual's rural or urban state
+        self._rural_urban_state = {
+            'True': 'Urban',
+            'False': 'Rural'
+        }
+
+        # define all properties that are categorised by rural or urban in addition to age and sex
+        self.cat_by_rural_urban_props = ['li_wealth', 'li_bmi', 'li_low_ex', 'li_ex_alc', 'li_wood_burn_stove',
+                                         'li_unimproved_sanitation',
+                                         'li_no_clean_drinking_water']
+
+        # date-stamp to label log files and any other outputs
+        self.datestamp: str = datetime.date.today().strftime("__%Y_%m_%d")
+
+        # a dictionary for gender descriptions. to be used when plotting by gender
+        self.gender_des: Dict[str, str] = {'M': 'Males', 'F': 'Females'}
+
+        # get all logs
+        self.all_logs = logs
+
+        # store un flattened logs
+        self.dfs = self.construct_dfs(self.all_logs)
+
+        self.outputpath = Path(path)  # folder for convenience of storing outputs
+
+    def construct_dfs(self, lifestyle_log) -> dict:
+        """ Create dict of pd.DataFrames containing counts of different lifestyle properties by date, sex and
+        age-group """
+        return {
+            k: unflatten_flattened_multi_index_in_logging(v.set_index('date'))
+            for k, v in lifestyle_log.items() if k in self.en_props.keys()
+        }
+
+    def custom_axis_formatter(self, df: pd.DataFrame, ax, li_property: str):
+        """
+        create a custom date formatter since the default pandas date formatter works well with line graphs. see an
+        accepted solution to issue https://stackoverflow.com/questions/30133280/pandas-bar-plot-changes-date-format
+
+        :param df: pandas dataframe or series
+        :param ax: matplotlib AxesSubplot object
+        :param li_property: one of the lifestyle properties
+        """
+        # make the tick labels empty so the labels don't get too crowded
+        tick_labels = [''] * len(df.index)
+        # Every 12th tick label includes the year
+        tick_labels[::12] = [item.strftime('%Y') for item in df.index[::12]]
+        ax.xaxis.set_major_formatter(ticker.FixedFormatter(tick_labels))
+        ax.legend(self.categories_desc[li_property] if li_property in self.categories_desc.keys()
+                  else [self.en_props[li_property][0]], loc='upper center')
+        # else [self.en_props[li_property]], bbox_to_anchor=(0.5, -0.27), loc='upper center')
+        plt.gcf().autofmt_xdate()
+
+    # 1. PLOT BY GENDER
+    # --------------------------------------------------------------------------------------------------------
+    def plot_categorical_properties_by_gender(self, li_property: str, categories: list):
```

Comment:
This method is on the long side and has quite a lot of repetition in it. There is a common shared structure across the various `if` conditions which we could factor out and pass in a function which deals with the property specific plotting, that is something like

```Python
def plot_categorical_property_by_gender(self, li_property, plot_function, n_rows, n_cols):
    fig, axes = plt.subplots(nrows=n_rows, ncols=n_cols, figsize=(10, 5))
    plot_function(axes, li_property)
    add_footnote(f'{self.en_props[li_property][1]}')
    fig.tight_layout()
    fig.savefig(self.outputpath / (li_property + self.datestamp + '.png'), format='png')
```